### ISSUE TITLE
Research spatial search algorithms to improve performance

### ISSUE THREAD

**Issue by jasonwebb** [jasonwebb/2d-differential-growth-experiments#4]
One critical aspect of this system that may not be handled by a physics engine is the ability to locate all of the nodes around a given node within a certain radius. All of these nodes will exert a "repulsion" force on each other, which would be easily handled by the physics engine, but _finding_ those nodes in the first place is a more specialized problem.

Various algorithms (and libraries) exist to efficiently solve this problem. These are generally classified as "spatial search" algorithms, and include algorithms like R-trees, KD trees, quadtrees, octrees, and maybe more. Some are more abstracted than others, and may or may not be appropriate for this simplified problem (a two-dimensional system)

**Comment by jasonwebb** [comment]
Added rbush + rbush-knn, but performance still isn't as great as I'd like. Will try other algorithms (like quadtree) soon.

**Comment by jasonwebb** [comment]
I have a feeling that my performance issues have more to do with my naive node injection algorithm than anything else. Working on that now.

Also I realized that I was storing [x,y] coordinates within local vectors in the Node class, not as top-level members of the Node class. I had to use a custom predicate function for knn to work, and I wonder if that could be a performance problem too. Will try moving them out of the vector class next.
